ray-operator/controllers/ray/raycluster_controller_unit_test.go

```diff
@@ -1742,15 +1738,97 @@ func TestCalculateStatus(t *testing.T) {
 	r.Client = fakeClient
 	newInstance, _ = r.calculateStatus(ctx, testRayCluster, nil)
 	assert.True(t, meta.IsStatusConditionPresentAndEqual(newInstance.Status.Conditions, string(rayv1.HeadPodReady), metav1.ConditionFalse))
-	condition = meta.FindStatusCondition(newInstance.Status.Conditions, string(rayv1.HeadPodReady))
-	assert.Equal(t, metav1.ConditionFalse, condition.Status)
 
 	// Test reconcilePodsErr with the feature gate enabled
 	newInstance, err = r.calculateStatus(ctx, testRayCluster, utils.ErrFailedCreateHeadPod)
 	assert.Nil(t, err)
 	assert.True(t, meta.IsStatusConditionPresentAndEqual(newInstance.Status.Conditions, string(rayv1.RayClusterReplicaFailure), metav1.ConditionTrue))
 }
 
+func TestRayClusterReadyCondition(t *testing.T) {
```

Comment:
After several tries, I decided to make it a separate function with TestCalculateStatus. Otherwise, it is too difficult to write with so many things coupled together. I think the test is fine being a little long, and these tests will be rewritten to use a real client in the future anyway.